packages/babel-generator/src/node/whitespace.js

```diff
@@ -80,9 +80,10 @@ export const nodes = {
    * Test if SwitchCase needs whitespace.
    */
 
-  SwitchCase(node: Object, parent: Object): ?WhitespaceObject {
+  SwitchCase(node: Object, parent: Object): WhitespaceObject {
     return {
       before: node.consequent.length || parent.cases[0] === node,
+      after: !node.consequent.length && parent.cases[parent.cases.length - 1] === node,
```

Comment:
Oh, i read this as two `!node.consequent.length && last(node.consequent)`. Never mind.